spec/rubocop/cop/lint/void_spec.rb

```diff
@@ -614,6 +614,18 @@ def foo=(rhs)
     RUBY
   end
 
+  it 'does not register `#each` block with conditional expressions that has multiple statements' do
+    expect_no_offenses(<<~RUBY)
+      enumerator_as_filter.each do |item|
```

Comment:
Won't this be different for this different enumerators, though? Maybe some Allowed list would make more sense than just assuming that multiple expressions are OK all the time.

Reply:
Initially, I also considered Allowed list approach, but it appears that the issue was simply that the following logic for `on_block` was not accounted for in the implementation of `on_begin`.
https://github.com/rubocop/rubocop/blob/v1.60.2/lib/rubocop/cop/lint/void.rb#L86

So, I think this approach could provide a more generic solution instead of Allowed list.